Add Conda build/test files for Symbiyosys and Boolector/Z3 solvers

mballance opened the pull request:
This PR adds support for building Symbiyosys on Linux platforms. 
I'm aware of one open issue that I could use some advice with: obtaining a version. I used the Verilator meta.yaml file as a basis for the Symbiyosys one, but the Verilator version mechanism doesn't work well for Symbiyosys. Any advice on typical approaches?

mithro commented on formal/symbiyosys/build.sh:
```diff
@@ -0,0 +1,105 @@
+
+# ilingeling.c contains a variable named 'clone'
+# This conflicts with the clone function, so we rename it
+sed -i -e 's%cd ${LINGELING_DIR}%cd ${LINGELING_DIR}\nsed -i -e "s/\\<clone\\>/_clone/g" ilingeling.c%' contrib/setup-lingeling.sh
```
This should probably just be a patch?

mithro commented on formal/symbiyosys/build.sh:
```diff
@@ -0,0 +1,105 @@
+#********************************************************************
+cd ${cwd}
+
+git clone https://github.com/Z3Prover/z3.git z3
```
This should be part of the meta.yaml?

mithro replied:
See https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#source-from-multiple-sources

mballance replied:
Nice! Somehow, even after searching I didn't manage to find that description of how to use multiple sources.

mballance replied:
Moved this to meta.yaml

mithro commented on formal/symbiyosys/build.sh:
```diff
@@ -0,0 +1,105 @@
+# CFLAGS=`echo ${CFLAGS_ORIG} | sed -e 's/-O[0-9]//g'`
+unset CFLAGS
+
+git clone https://github.com/Boolector/boolector.git
```
This should be part of the meta.yaml

mballance replied:
moved to meta.yaml

mballance commented:
> Is the issue a missing tag?

Most likely. The SymbiYosys repository has no tags, no releases, and no non-master branches. Is there an accepted way to give a version in this case? I guess we could use the Git commit hash. Would that play okay with Conda's expectations?

mithro commented:
> > Is the issue a missing tag?
> 
> Most likely. The SymbiYosys repository has no tags, no releases, and no non-master branches. Is there an accepted way to give a version in this case? I guess we could use the Git commit hash. Would that play okay with Conda's expectations?

Clone the repository to a seperate location and add a v0.0 tag to the first ever commit. Then use something like https://github.com/apps/pull to keep the repository in sync with the upstream.

mithro commented:
Friendly ping!

mballance commented:
Hey Tim, Thanks for the reminder -- getting these issues address sort of got lost in the shuffle. I think the latest commit addresses your feedback. I've changed to specifying multiple sources, dealt with the version issue, and created a separate patch file.

mithro commented:
@mballance You will want to add `formal/symbiyosys` to the `.travis.yml` file too.

mballance commented:
Good point, @mithro. I think I got the .travis.yaml file right. I also added an entry to the README

mithro commented:
@mballance Did you forget to push?

mballance commented:
Oops... Accidentally pushed to the wrong branch. You should have the changes now

mithro commented:
LGTM.
